frontend/src/scenes/data-warehouse/externalDataSourcesLogic.ts

```diff
@@ -62,3 +62,218 @@ export const externalDataSourcesLogic = kea<externalDataSourcesLogicType>([
         },
     })),
 ])
+
+export const fetchExternalDataSourceJobs = async (
```

Comment:
All of this logic needs to be in kea. This might work in the hook pattern but it's significantly diverging from how state is managed across posthog and will get really difficult to reason about